### PR TITLE
Enable tracing

### PR DESCRIPTION
**Pull request by dargilco**
* Update method .diagnostics.get_connection_string()
* Update method .diagnostics.enable()
* Add Inference samples with console and Azure Monitor tracing. Console tracing works for both, Azure Monitor works for azure-ai-inference client, but not for openai client. Still investigating this.
* Still need to add samples for tracing to OTLP collector endpoint.
* Add some diagnostic tests. Still some work to be done. 

Per discussion with Liudmila, I will change the operation group from ".diagnostics" to ".telemetry". I will do this in a 2nd PR, in order not to pollute this one.

### CODE REVIEW

File: sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py [hunk 1]
```diff
@@ -310,43 +312,50 @@ async def list(
...
-                raise ValueError("Application Insights resource does not have a connection string")
+                self._connection_string = app_insights_respose.properties.connection_string
 
-            self.connection_string = app_insights_respose.properties.connection_string
+        self._get_connection_string_called = True
+        return self._connection_string
 
-        return True
+
+    # TODO: what about `set AZURE_TRACING_GEN_AI_CONTENT_RECORDING_ENABLED=true`?
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py:349]: Checking if these TODO comments are still valid?

File: sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py [hunk 2]
```diff
@@ -310,43 +312,50 @@ async def list(
...
-    async def enable(self, **kwargs) -> bool:
-        """Enable Application Insights tracing.
-        This method makes service calls to get the properties of the Applications Insights resource
-        connected to the Azure AI Studio Project. If Application Insights was not enabled for this project,
-        this method will return False. Otherwise, it will return True. In this case the Application Insights
-        connection string can be accessed via the `.diagnostics.connection_string` property.
-
-        :return: True if Application Insights tracing was enabled. False otherwise.
-        :rtype: bool
+    async def get_connection_string(self) -> str:
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py:322]: could there be other connections that app insights? Wondering if the name should be e.g. get_app_insights_connection.

File: sdk/ai/azure-ai-projects/samples/inference/sample_chat_completions_with_azure_ai_inference_client_and_console_tracing.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    * PROJECT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+    * AZURE_TRACING_GEN_AI_CONTENT_RECORDING_ENABLED=true - Optional. For detailed traces, including chat request and response messages.
+"""
+import os
+import sys
+from azure.ai.projects import AIProjectClient
+from azure.ai.inference.models import UserMessage
+from azure.identity import DefaultAzureCredential
+#from azure.monitor.opentelemetry import configure_azure_monitor
+#from azure.ai.inference.tracing import AIInferenceInstrumentor
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/samples/inference/sample_chat_completions_with_azure_ai_inference_client_and_console_tracing.py:29]: remove?

**Review by jhakulin — APPROVED** [review]

File: sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py (hunk 1 above)
Comment (dargilco) [reply sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py:349]: Yes, thanks, still valid. I need to discuss this further with Dan, Johan, Liudmila and Trupti.

File: sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py (hunk 2 above)
Comment (dargilco) [reply sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py:322]: I debated this... I originally had it as `get_application_insights_connection_string()` and it seemed too long. Let me get feedback on this and make a change in a separate PR if needed.

**Pull request merged**